mycore-base/src/main/java/org/mycore/common/xml/MCREntityResolver.java

```diff
@@ -104,15 +108,29 @@ private static boolean isAbsoluteURL(String url) {
 
     private InputSource resolveEntity(String publicId, String systemId,
         MCRThrowFunction<CatalogEntityIdentifier, InputSource, IOException> alternative) throws IOException {
+        InputSource entity = null;
         try {
-            InputSource entity = catalogResolver.resolveEntity(publicId, systemId);
+            entity = catalogResolver.resolveEntity(publicId, systemId);
             if (entity != null) {
                 return resolvedEntity(entity);
             }
         } catch (CatalogException e) {
             LOGGER.debug(e.getMessage());
         }
-        return alternative.apply(new CatalogEntityIdentifier(publicId, systemId));
+        entity = alternative.apply(new CatalogEntityIdentifier(publicId, systemId));
+        if (entity == null) {
+            MCRURIResolver uriResolver = MCRURIResolver.instance();
+            String scheme = uriResolver.getScheme(systemId, null);
+            try {
+                if (scheme != null && uriResolver.getResolver(scheme) != null) {
+                    Source s = MCRURIResolver.instance().resolve(systemId, null, false);
+                    return new MCRSourceContent(s).getInputSource();
+                }
+            } catch (MCRUsageException | TransformerException e) {
+                return null;
+            }
```

Comment:
Simply returning null without any notice? Could cause hard to find errors. We should at least **warn** in the logs.

Reply:
O.K. I will add a log message, but the old implementation did also return silent null